js/copy-code.js

```diff
@@ -0,0 +1,123 @@
+
+  return wrapper;
+};
+
+/**
+ * Creates a copy button component.
+ *
+ * @return {HTMLButtonElement} - A button for copying code `copy-code__button`.
+ */
+const createCopyButton = () => {
+  const btn = document.createElement("button");
+  const btnTxt = document.createElement("span");
+
+  btnTxt.setAttribute("aria-hidden", "true");
+  btnTxt.textContent = "Copy";
+  btn.append(btnTxt);
+
+  btn.className = COPY_BUTTON_CLASS;
+  btn.setAttribute("type", "button");
+
+  return btn;
+};
+
+/**
+ * Creates a screen reader only div element for success text
+ *
+ * @return {HTMLDivElement} - A div for screen reader text
+ */
+const createSRStatus = () => {
+  const srText = document.createElement("div");
+  srText.classList.add("usa-sr-only", SR_STATUS_MESSAGE_CLASS);
+  srText.textContent = SR_DEFAULT_MESSAGE;
+
+  return srText;
+};
+
+/**
+ * Copy <code> text content when copy code button is clicked.
+ *
+ * @param {Event} event - The click event from copy code button.
+ * @return {Clipboard} - Code element contents written to clipboard.
+ */
+const copyOnClick = (event) => {
+  const copyBtn = event.currentTarget;
+  const btnWrapper = copyBtn.closest(COPY_WRAPPER);
+  const labelVisual = copyBtn.querySelector("[aria-hidden]");
+  const labelSR = btnWrapper.querySelector(SR_STATUS_MESSAGE);
+
+  // Set success state
+  copyBtn.classList.add(COPY_BUTTON_SUCCESS_CLASS);
+  labelVisual.textContent = "Copied!";
+  labelSR.textContent = SR_SUCCESS_MESSAGE;
+
+  // Set aria-live once _after_ initial status update.
+  // Prevents VoiceOver from reading "Copy code" when parent accordion expands.
+  //
+  // See first item in PR comment:
+  // https://github.com/uswds/uswds-site/pull/2059#pullrequestreview-1503414868
+  if (!labelSR.hasAttribute("aria-live")) {
+    labelSR.setAttribute("aria-live", "polite");
+  }
+
+  // After timeout, reset to default state
+  setTimeout(() => {
+    copyBtn.classList.remove(COPY_BUTTON_SUCCESS_CLASS);
+    labelVisual.textContent = "Copy";
+    labelSR.textContent = SR_DEFAULT_MESSAGE;
+  }, 3000);
+
+  // Select section code and copy to clipboard
+  const copyComponent = copyBtn.closest(`.${COPY_CODE_CLASS}`);
+  const codeElement = copyComponent.querySelector("code");
+  return navigator.clipboard.writeText(codeElement.textContent);
+};
+
+/**
+ * Initialize code component.
+ *
+ * Iterate through all code components and attach dynamically generated
+ * button & wrapper to DOM. Copy code element contents when button is clicked.
+ *
+ */
+const init = () => {
+  COPY_CODE.forEach((copyCodeElement) => {
+    const COPY_CODE_ID = copyCodeElement.id;
+    const SCREEN_READER_ID = `${COPY_CODE_ID}__sr-label`;
+
+    const copyWrapper = createWrapper();
+    const copyButton = createCopyButton();
+    const labelSROnly = createSRStatus();
+
+    labelSROnly.setAttribute("id", SCREEN_READER_ID);
+    copyButton.setAttribute("aria-labelledby", SCREEN_READER_ID);
+
+    copyWrapper.append(labelSROnly, copyButton);
```

Comment:
Consolidated these two elements (`labelSrOnly` and `copyButton`) into a single `append` in [841ae6e](https://github.com/uswds/uswds-site/pull/2059/commits/841ae6ebf685edd487d40d585eed2244c35f8533).

```diff
-    copyWrapper.appendChild(labelSROnly);
-    copyWrapper.appendChild(copyButton);
+    copyWrapper.append(labelSROnly, copyButton);
```